### PR TITLE
Extract util method for row moving commands

### PR DESCRIPTION
**Pull request by jomarko**
@manstis please have a look, and also question, during moving, there is `oldIndex` and `newIndex`, during moving multiple rows, is the `newIndex` index for first or last of moved rows?

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package org.kie.workbench.common.dmn.client.commands.util;
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java]: License please.

**Comment by manstis** [comment]
Yay.. I broke it by sync'ing my fork with ```kiegroup/kie-wb-common``` Sorry.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java (hunk 1 above)
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java]: As usually :D will do

**Comment by jomarko** [comment]
@manstis I am definitely for the check if the moved rows are contiguous block, it sounds to me like candidate for another util method.

Back to my question about moving multiple rows and the `newIndex`. According your answer I think the test method `testMoveTwoRowsDown` from this PR shows that the current implementation is not correct, what do you think?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    @Test
+    public void testMoveTwoRowsUp() throws Exception {
+        CommandUtils.moveRows(allRows, Arrays.asList(decisionRuleTwo, decisionRuleThree), 1, 0);
+
+        Assertions.assertThat(allRows).containsSequence(decisionRuleTwo, decisionRuleThree, decisionRuleOne);
+    }
+
+    @Test
+    public void testMoveTwoRowsDown() throws Exception {
+        CommandUtils.moveRows(allRows, Arrays.asList(decisionRuleOne, decisionRuleTwo), 0, 2);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java]: It's not entirely plausible to move two rows to index 2. Index 2 is the last row in the table and only has space for 1 row.

I therefore think the results (assuming this test passes!) ```decisionRuleThree```, ```decisionRuleOne```, ```decisionRuleTwo``` are correct. This gives the impression that "new index" is that of the last row; but the result would (should!) be the same as if you moved ```decisionRuleOne``` and ```decisionRuleTwo``` to "new index" 1.

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java]: Totally confused, I had the line like [1], but then the test was failing, please throw an action item what to do.

[1]
```
CommandUtils.moveRows(allRows, Arrays.asList(decisionRuleOne, decisionRuleTwo), 0, 1);
 
```

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java]: It's plausible something is broken; it's plausible the test is wrong.. I'll fetch and fiddle.

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtilsTest.java]: @jomarko OK, so I think the original implementation of the commands was flawed.

Changing ```CommandUtils.moveRows(..)``` to the follow makes the test [1] work and I did not experience any issues running the web app itself and moving rows (but might be worth double-checking). I must admit however it does make me a little nervous and the more _eyes_ reviewing the better! :
```
public static <T> void moveRows(final List<T> allRows, final List<T> rowsToMove, final int index) {
    allRows.removeAll(rowsToMove);
    allRows.addAll(index, rowsToMove);
}
```
[1] ```CommandUtils.moveRows(allRows, Arrays.asList(decisionRuleOne, decisionRuleTwo), 1)```

**Comment by jomarko** [comment]
@manstis I did related changes in appformer:
- https://github.com/manstis/appformer/pull/1

Ensemble should be ready for merge, I checked that rows can be moved in DMN authoring. Should I check some other components where could my changes broke the rows moving?

**Comment by manstis** [comment]
@jomarko It'd be good to ensure the Guided Decision Table Editor still works with merged/collapsed rows.

**Comment by jomarko** [comment]
@manstis I will have a look, stay tuned please.

**Comment by jomarko** [comment]
@manstis I checked manually and I have not spotted problem in GDT even with merged/collapsed cells, however if you point me to proper class I would add also some junit test.

**Comment by manstis** [comment]
@jomarko Hi, the tests of interest are probably in https://github.com/kiegroup/appformer/blob/master/uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/model/impl/GridRowsMergedTest.java

**Comment by jomarko** [comment]
Replaced by kiegroup/kie-wb-common#1339

**Pull request closed without merging**